PSFramework/internal/loggingProviders/gelf.provider.ps1

```diff
@@ -0,0 +1,159 @@
+﻿# Action that is performed on registration of the provider using Register-PSFLoggingProvider
+$registrationEvent = {
+
+}
+
+#region Logging Execution
+# Action that is performed when starting the logging script (or the very first time if enabled after launching the logging script)
+$begin_event = {
+    $gelf_gelfserver = Get-PSFConfigValue -FullName 'PSFramework.Logging.GELF.GelfServer'
+    $gelf_port = Get-PSFConfigValue -FullName 'PSFramework.Logging.GELF.Port'
+    $gelf_encrypt = Get-PSFConfigValue -FullName 'PSFramework.Logging.GELF.Encrypt'
+
+    $gelf_paramSendPsgelfTcp = @{
+        'GelfServer' = $gelf_gelfserver
+        'Port' = $gelf_port
+        'Encrypt' = $gelf_encrypt
+    }
+}
+
+# Action that is performed at the beginning of each logging cycle
+$start_event = {
+    if (-not (Get-Module -Name 'PSGELF')) {
+        Import-Module -Name 'PSGELF'
+    }
+}
+
+# Action that is performed for each message item that is being logged
+$message_Event = {
+	Param (
+		$Message
+	)
+
+    $gelf_params = $gelf_paramSendPsgelfTcp.Clone()
+    $gelf_params['ShortMessage'] = $Message.Message
+    $gelf_params['HostName'] = $Message.ComputerName
+    $gelf_params['DateTime'] = $Message.Timestamp
+
+    $gelf_params['Level'] = switch ($Message.Level) {
+        'Critical' { 1 }
+        'Important' { 1 }
+        'Output' { 3 }
+        'Host' { 4 }
+        'Significant' { 5 }
+        'VeryVerbose' { 6 }
+        'Verbose' { 6 }
+        'SomewhatVerbose' { 6 }
+        'System' { 6 }
+
+        default { 7 }
+    }
+
+    # build the additional fields
+    $gelf_properties = $Message.PSObject.Properties | Where-Object {
+        $_.Name -notin @('Message', 'ComputerName', 'Timestamp', 'Level')
+    }
+
+    $gelf_params['AdditionalField'] = @{'testing' = $gelf_params['Level'] }
+    foreach ($gelf_property in $gelf_properties) {
+        $gelf_params['AdditionalField'][$gelf_property.Name] = $gelf_property.Value
+    }
+
+    PSGELF\Send-PSGelfTCP @gelf_params
+}
+
+# Action that is performed for each error item that is being logged
+$error_Event = {
+	Param (
+		$ErrorItem
+	)
```

Comment:
Not logging the error objects?
Exceptions might really help with debugging code.
It's possible to compress them into a single string object using `ConvertTo-PSFClixml` (which will then need `ConvertFrom-PSFClixml` in order to read it, of course)

Reply:
I copied off the logfile provider and didn't bother to look at the filesystem provider. Graylog is big on JSON, so I will probably convert to JSON - they have extractors to pull the info out for you (pretty neat). 

When would this be called? When ErrorRecord is provided in Write-PSFMessage?

Reply:
This is called when ...

 - Providing `-ErrorRecord` on `Write-PSFMessage`
 - Calling `Stop-PSFFunction`

In both instances it logs a message (generally a warning message), which is the same event as all the others. It _also_ logs an error event, that has details on any exceptions.
To inspect the object it receives, you can call `Get-PSFMessage -Errors` after logging one of those.

Reply:
At the moment, this is unfortunately not possible at all.
The reason for that is that - for historical reasons - the two objects are queued separately and the messages get processed in bulk, then the errors get processed in bulk. You could compare timestamps, which would make it correct, but that would require gathering all events and writing them during the end block (which I advice against).

Alright, I'll put consolidating that into tomorrows coding sprint, so that the regular messages have access to the error object. Should have been one that way long since, but since I used to be the only person messing with logging providers, I never had an external kick to the butt to prioritize it ^^

Reply:
Having trouble now - it seems like Get-PSFConfigValue is not defined when the start event is called.

```powershell
(Get-PSFLoggingProvider -Name gelf).Errors
```

```
Exception calling "InvokeScript" with "4" argument(s): "The term 'Get-PSFConfigValue' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name,
or if a path was included, verify that the path is correct and try again."
At line:32 char:11
+ ...       try { $ExecutionContext.InvokeCommand.InvokeScript($false, ([Sy ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : CommandNotFoundException
```

Reply:
Hm, interesting.
Do you run this on a system that does not have the PSFramework installed in a known location ($env:PSModulePath)?
That should not be an issue, of course, but it currently is (because the logging script that marshals the providers does not automatically pick up the path from where the module was imported - need to add automatically picking up the same version of the module that started it - to make sure it uses just the same version as the main runspace - adding an issue to add that to the next version)

Reply:
That did the trick! Updating the PR with the requested changes now.